### ISSUE TITLE
[rfc] Improve SLURM + Ray experience

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#9043]
### Describe your feature request

Is anyone here having trouble using SLURM with Ray? 

We'd love to hear your experience (either good or bad).

For starters, two things to do that might help are:
1. Provide links to other SLURM templates using Ray
2. Increase the starter SLURM template to 30 seconds wait time.

Questions/comments are welcome!

**Comment by pengzhenghao** [comment]
Nice to see you guys paying attention to slurm experience! I personally using a template txt file and a brief python script to help invoking a brunch of ray programs in slurm cluster. The idea is easy, just add some placeholders to the suggested slurm script in document and replace them by python script.

https://gist.github.com/pengzhenghao/b348db1075101a9b986c4cdfea13dcd6

There are two problems that I have encounter:

1. You don't know whether the things is working well. I usally have to wait for 5~10 minuts to see the first table printed by tune.
2. slurm supports direct communication between nodes WITHIN one partition, but the communication ACROSS partitions or ACROSS clusters is doubtful. I don't try this out, but this is the concern of my coworker and I don't know the answer.

**Comment by pengyuan-zhou** [comment]
Yes, please check #9028 
BR,
Pengyuan

**Comment by yutaizhou** [comment]
I support increasing the wait time for the starter SLURM template. 5s has definitely not been enough in my experience. I think a minimum is 20s but again, that's just on our SLURM system. 

It certainly hasn't been easy using Ray on SLURM, given the limited documentation on it and the small community. Our HPEC admins and my project's engineers had to add a good amount of modification to the template to make it work. The two main problems we ran into were

1. The 5s timing being too short
2. Lack of support for non-base-10 GPU ID naming (https://github.com/ray-project/ray/issues/8313)

We were able to solve both of these issues, but again, we didn't get much support since the SLURM use case is rather rare. 

Additionally, for some reason our non-actor print messages don't even get printed out until after the job is done. This could be something on our end, but I didn't seen this behavior when I ran the same program on a local non-SLURM machine. 

We are also having trouble fully utilizing multiple nodes for our jobs. I sent the problem to @richard4912 over slack but haven't had the time to work on it since.
